packages/farrow-api-server/src/apiResponse.ts

```diff
@@ -2,12 +2,27 @@ import type { JsonType } from 'farrow-schema'
 
 export { JsonType }
 
+export type IntrospectionCalling = {
+  input: {
+    __introspection__: true
+  }
+}
+
+export type SingleCalling = {
+  path: string[]
+  input: Readonly<JsonType>
+}
+
+export type BatchCalling = {
+  __batch__: true
+  callings: Readonly<SingleCalling[]>
+}
+
+export type Calling = SingleCalling | BatchCalling | IntrospectionCalling
```

Comment:
It should use `tagged-unions`.

```typescript
// __introspection__ = true is no need since we have tag now.
export type IntrospectionCalling = {
  type: 'IntrospectionCalling';
};

export type SingleCalling = {
  type: 'SingleCalling';
  path: string[];
  input: Readonly<JsonType>;
};

// __batch__ = true is no need since we have tag now
export type BatchCalling = {
  type: 'BatchCalling';
  callings: Readonly<SingleCalling[]>;
};

export type Calling = SingleCalling | BatchCalling | IntrospectionCalling
```